lang/cs/Org.Apache.REEF.Wake/Time/Time.cs

```diff
@@ -50,33 +55,21 @@ public override bool Equals(object obj)
             {
                 return true;
             }
-            Time other = obj as Time;
+            
+            // Note: This allows for the very strange semantic where
+            // two different subclasses might return true for Equals.
+            var other = obj as Time;
```

Comment:
Perhaps we should have an `ITime` interface and an `IComparer<ITime>` class instead?